core/block_validator.go

```diff
@@ -65,6 +65,11 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash {
 		return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash)
 	}
+	if block.WithdrawalsHash() != nil {
```

Comment:
```suggestion
	if header.WithdrawalsHash != nil {
```
On the line below, you access it directly via `header.WithdrawalsHash`, so I don't see why you access it via `block` on this line (the access via block also causes an extra copy).  Writing it like this makes it more consistent and easy to follow IMO